### PR TITLE
Ability to Override Secret Backend Using Environment Variable

### PR DESCRIPTION
**Pull request by ericreeves**
This simple PR adds the ability to override the default secret backend of "keychain" using the environment variable AWS_VAULT_PL_BACKEND.

### CODE REVIEW

File: zsh-aws-vault.plugin.zsh [hunk 1]
```diff
@@ -5,39 +5,40 @@ AWS_VAULT_PL_DEFAULT_PROFILE=${AWS_VAULT_PL_DEFAULT_PROFILE:-default}
...
-alias avs='aws-vault server'
+alias av='aws-vault --backend=${AWS_VAULT_PL_BACKEND}'
+alias avs='aws-vault --backend=${AWS_VAULT_PL_BACKEND} server'
 
 #--------------------------------------------------------------------#
 # Convenience Functions                                              #
 #--------------------------------------------------------------------#
 function avl() {
-  aws-vault login $@
+  aws-vault --backend=${AWS_VAULT_PL_BACKEND} login $@
```
Comment (blimmer) [inline zsh-aws-vault.plugin.zsh:20]: I wonder, could you use the av alias you defined above to reduce needing to duplicate the --backend part throughout?

Comment (ericreeves) [reply zsh-aws-vault.plugin.zsh:20]: That's a fantastic idea.  I'll update the PR.

**Comment by blimmer** [comment]
Actually, I just got back to my computer and was able to look into this more closely. Does this plugin need to provide this behavior? It looks like there's a top-level `AWS_VAULT_BACKEND` environment variable the user can set to accomplish this. 

I think `keychain` is probably only valid for macOS, so this default might not work for some linux users and would require out-of-the-box configuration.

**Comment by ericreeves** [comment]
I didn't see this in the aws-vault help, and totally neglected to check the usage docs at the Github repo.  It's totally not necessary.

Cheers!

**Comment by blimmer** [comment]
No worries!

**Pull request closed without merging**